service/resource/endpoint/resource.go

```diff
@@ -64,6 +65,42 @@ func containsIP(ips []string, ip string) bool {
 	return false
 }
 
+func constructK8sEndpoint(endpoint Endpoint, k8sClient kubernetes.Interface) (*apiv1.Endpoints, error) {
```

Comment:
I moved this to resource as it's used while deleting and creating.

Reply:
Nitpick: in go all "construction" methods are prefixed with `new`.

Reply:
I renamed this and reordered it.